tools/downloader/topologies.py

```diff
@@ -0,0 +1,233 @@
+import sys
+import os
+import re
+import yaml
+import types
+
+import common
+
+if 'INTEL_OPENVINO_DIR' in os.environ:  # As a part of OpenVINO
+    _loc = os.path.join(os.environ['INTEL_OPENVINO_DIR'], 'deployment_tools', 'open_model_zoo')
+    _modelsDir = os.path.join(_loc, 'models')
+    _downloader = os.path.join(_loc, 'tools', 'downloader', 'downloader.py')
+else:  # Standalone
+    _loc = os.path.dirname(__file__)
+    _modelsDir = os.path.join(_loc, '..', '..', 'models')
+    _downloader = os.path.join(_loc, 'downloader.py')
+
+_precisionMarker = '$precision'
+_cache = os.path.abspath(os.getenv('OPENCV_OPEN_MODEL_ZOO_CACHE_DIR', 'models'))
+
+# Parse topologies.
+for _moduleName in os.listdir(_modelsDir):
+    _modelsSubDir = os.path.join(_modelsDir, _moduleName)
+    if not os.path.isdir(_modelsSubDir):
+        continue
+
+    _module = types.ModuleType(_moduleName)
+    sys.modules['topologies.' + _moduleName] = _module
+    globals()[_moduleName] = _module
+    _module.__dict__['sys'] = sys
+    _module.__dict__['os'] = os
+
+    # List of models which have versions (mostly DLDT models). For such kind of models
+    # with the highest version we will add short names. In example, for
+    # "face_detection_retail_0004" and "face_detection_retail_0005" method with name
+    # "face_detection_retail" returns result of "face_detection_retail_0005".
+    _versionedNames = {}
+
+    for _topologyName in os.listdir(_modelsSubDir):
+        _config = os.path.join(_modelsSubDir, _topologyName, 'model.yml')
```

Comment:
This still duplicates the config parsing logic from `common.py`. You should reuse that instead, or use the info dumper.

Reply:
`common.load_topologies` reused